### ISSUE TITLE
TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by lucas03** [agronholm/sphinx-autodoc-typehints#66]
Hi, I encountered this issue when building docs and included sphinx-autodoc-typehints:

`sphinx-build -M html docs docs/_build -q 2>&1`

```
Exception occurred:
  File "/usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py", line 122, in format_annotation
    extra = '\\[{}]'.format(', '.join(format_annotation(param) for param in params))
TypeError: 'NoneType' object is not iterable
The full traceback has been saved in /tmp/sphinx-err-yvc80trd.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
/srv/backend/booking # nano /tmp/sphinx-err-yvc80trd.log
```

```
# Sphinx version: 1.8.2
# Python version: 3.6.7 (CPython)
# Docutils version: 0.14
# Jinja2 version: 2.10
# Last messages:
#   reading sources... [  6%] kw/kw.backoff.models.additional_services
#
#   reading sources... [  7%] kw/kw.mambo
#
#   reading sources... [  8%] kw/kw.mambo.constants
#
#   reading sources... [  9%] kw/kw.mambo.handlers
#
#   reading sources... [  9%] kw/kw.mambo.models
#
# Loaded extensions:
#   sphinx.ext.mathjax (1.8.2) from /usr/local/lib/python3.6/site-packages/sphinx/ext/mathjax.py
#   alabaster (0.7.12) from /usr/local/lib/python3.6/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc (1.8.2) from /usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx_autodoc_typehints (unknown version) from /usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py
#   sphinx.ext.viewcode (1.8.2) from /usr/local/lib/python3.6/site-packages/sphinx/ext/viewcode.py
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/sphinx/cmd/build.py", line 304, in build_main
    app.build(args.force_all, filenames)
  File "/usr/local/lib/python3.6/site-packages/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "/usr/local/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "/usr/local/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 360, in build
    updated_docnames = set(self.read())
  File "/usr/local/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 468, in read
    self._read_serial(docnames)
  File "/usr/local/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 490, in _read_serial
    self.read_doc(docname)
  File "/usr/local/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 534, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/usr/local/lib/python3.6/site-packages/sphinx/io.py", line 318, in read_doc
    pub.publish()
  File "/usr/local/lib/python3.6/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/local/lib/python3.6/site-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
...
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/directive.py", line 131, in run
    documenter.generate(more_content=self.content)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 819, in generate
    self.document_members(all_members)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 740, in document_members
    check_module=members_check_module and not isattr)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 1247, in generate
    all_members=all_members)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 816, in generate
    self.add_content(more_content)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 1229, in add_content
    ModuleLevelDocumenter.add_content(self, more_content)
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 554, in add_content
    for i, line in enumerate(self.process_doc(docstrings)):
  File "/usr/local/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 514, in process_doc
    self.options, docstringlines)
  File "/usr/local/lib/python3.6/site-packages/sphinx/application.py", line 510, in emit
    return self.events.emit(event, self, *args)
  File "/usr/local/lib/python3.6/site-packages/sphinx/events.py", line 80, in emit
    results.append(callback(*args))
  File "/usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py", line 197, in process_docstring
    formatted_annotation = format_annotation(annotation)
  File "/usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py", line 106, in format_annotation
    extra = '\\[{}]'.format(', '.join(format_annotation(param) for param in params))
  File "/usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py", line 106, in <genexpr>
    extra = '\\[{}]'.format(', '.join(format_annotation(param) for param in params))
  File "/usr/local/lib/python3.6/site-packages/sphinx_autodoc_typehints.py", line 122, in format_annotation
    extra = '\\[{}]'.format(', '.join(format_annotation(param) for param in params))
TypeError: 'NoneType' object is not iterable
```

**Comment by bpeake-illuscio** [comment]
Created a PR with the fix here: https://github.com/agronholm/sphinx-autodoc-typehints/pull/67/files

**Comment by agronholm** [comment]
Fixed via 645fa6b9.

**Comment by lucas03** [comment]
thank you!
